src/service/task.rs

```diff
@@ -664,11 +664,10 @@ impl crate::server::MyService {
                 )))
             }
         }
-        self.check_user_consent(decision.user_consent.as_ref().unwrap(), core_public_key_vec)?;
+        let user_public_key_vec_from_userconsent =
```

Comment:
`user_public_key_vec_from_user_consent`? (I know it's long but it's only the last one...)